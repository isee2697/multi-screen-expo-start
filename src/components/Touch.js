import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { utilStyles } from '../api/constants';

class Touch extends React.PureComponent {
  render() {
    const { icon, opacityActive, onPress, style, text, textStyle } = this.props;

    return (
      <TouchableOpacity
        activeOpacity={opacityActive}
        onPress={onPress}
        style={style}
      >
        {icon && icon}
        <Text style={textStyle}>{text}</Text>
      </TouchableOpacity>
    );
  }
}

// default props
Touch.defaultProps = {
  icon: null,
  opacityActive: utilStyles.opacityActive,
  style: {},
  text: '',
  textStyle: {
    textAlign: 'center'
  }
};

// type checking
Touch.propTypes = {
  // required
  onPress: PropTypes.func.isRequired,

  // optional
  icon: PropTypes.element,
  opacityActive: PropTypes.number,
  style: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.number,
    PropTypes.object
  ]),
  text: PropTypes.string,
  textStyle: PropTypes.object
};

export default Touch;
