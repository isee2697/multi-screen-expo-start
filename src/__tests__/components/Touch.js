import React from 'react';
import { shallow } from 'enzyme';

import Touch from '../../components/Touch';

const component = shallow(<Touch onPress={() => jest.fn()} />);

describe('<Touch />', () => {
  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
