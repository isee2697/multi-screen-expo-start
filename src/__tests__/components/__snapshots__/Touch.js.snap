// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Touch /> renders correctly 1`] = `
<TouchableOpacity
  accessibilityComponentType="button"
  accessibilityLabel=""
  accessibilityTraits="button"
  accessible={true}
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#1d2d3a",
      "borderRadius": 80,
      "borderWidth": 0.5,
      "paddingHorizontal": 24,
      "paddingVertical": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#1d2d3a",
        "textAlign": "center",
      }
    }
  />
</TouchableOpacity>
`;
